module/epochs/epoch_lookup.go

```diff
@@ -39,6 +37,23 @@ func (cache *epochRangeCache) latest() epochRange {
 	return cache[2]
 }
 
+// extendLatestEpoch updates the final view of the latest epoch with the final view of the epoch extension.
+// No errors are expected during normal operation.
+func (cache *epochRangeCache) extendLatestEpoch(extensionFinalView uint64) error {
+	// sanity check: latest epoch should already be cached.
+	if !cache[2].exists() {
+		return fmt.Errorf("sanity check failed: latest epoch does not exist")
+	}
+
+	// sanity check: extensionFinalView should be greater than final view of latest epoch
+	if cache[2].finalView > extensionFinalView {
+		return fmt.Errorf("sanity check failed: latest epoch final view %d greater than extension final view %d", cache[2].finalView, extensionFinalView)
+	}
+
+	cache[2].finalView = extensionFinalView
```

Comment:
```suggestion
	// duplicate events are no-ops
	if cache[2].finalView == extensionFinalView {
		return nil
	}	
	
	cache[2].finalView = extensionFinalView
```

Behaviour-wise I think the current implementation correctly handles duplicate events. If `extensionFinalView == cache[2].finalView`, we will get to this line of code but it will be a no-op. Personally I think it's beneficial to add a specific conditional for the duplicate event case to make clear that dupes are handled and how (even though it's functionally the same).